crossbar/router/auth/cryptosign.py

```diff
@@ -254,10 +254,8 @@ def hello(self, realm, details):
             if attr not in extra:
                 return types.Deny(message='missing required attribute {}'.format(attr))
 
+        # enforce the backend to be on the same realm as actual client
         realm = extra['proxy_realm']
-        details.authid = extra['proxy_authid']
-        details.authrole = extra['proxy_authrole']
-        details.authextra = extra.get('proxy_authextra', None)
```

Comment:
The realm should already match (right? because here https://github.com/crossbario/crossbar/blob/master/crossbar/worker/proxy.py#L291).

As for authid / authrole .. hrm. We have "dynamic" versus "static" config cases .. for "dynamic", it's probably *least* surprising if there's no magic re-assignment -- we just pass through what the dynamic auth returns (and it has to return an authid + authrole).

For the static case, I guess it comes down to what the config-file should look like. If we re-assign *only-if* the incoming authid/authrole are `None` (I just pushed it like that) then config could look like this:

```
                    "auth": {
                        "cryptosign-proxy": {
                            "type": "static",
                            "principals": {
                                "user1": {
                                    "role": "anonymous",
                                    "authorized_keys": [
                                        "a1fd4c2c2954b92ef784b4d14442e2eb159cc74040bb59e43e84b3c56719256f"
                                    ]
                                },
                                "user2": {
                                    "role": "foo",
                                    "authorized_keys": [                                                              
                                        "a1fd4c2c2954b92ef784b4d14442e2eb159cc74040bb59e43e84b3c56719256f"
                                    ]
                                }
                            }
                        }
                    }
```

...but that might bet a little weird, as you'd basically have to replicate ALL your frontend users, and they'll have the same set of authorized keys (all your nodes). If we have the proxy->backend config put in the node-id as `authid` then the config could look more like this:

```
                   "auth": {
                        "cryptosign-proxy": {
                            "type": "static",
                            "principals": {
                                "node1": {
                                    "role": "anonymous",
                                    "authorized_keys": [
                                        "a1fd4c2c2954b92ef784b4d14442e2eb159cc74040bb59e43e84b3c56719256f"            
                                    ]
                                }
                            }
                        }
                    }
```

If you want to assign different roles per-user (real frontend user) then you'd need to move to dynamic auth. To me it "feels better" to have just node-names and node-pubkeys burned into static configu .. but that's maybe not a good argument either way ;)